projects.go

```diff
@@ -464,59 +465,85 @@ func (s *ProjectsService) GetProjectEvents(pid interface{}, opt *GetProjectEvent
 	return p, resp, err
 }
 
+// ContainerExpirationPolicy represents the container expiration attributes for a project.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/projects.html
+type ContainerExpirationPolicy struct {
+	Cadence         string     `json:"cadence"`
+	KeepN           int        `json:"keep_n"`
+	OlderThan       string     `json:"older_than"`
+	NameRegexDelete string     `json:"name_regex_delete"`
+	NameRegexKeep   string     `json:"name_regex_keep"`
+	Enabled         bool       `json:"enabled"`
+	NextRunAt       *time.Time `json:"next_run_at"`
+}
+
+// ContainerExpirationPolicyAttributes represents the container expiration attributes for a project.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/projects.html
+type ContainerExpirationPolicyAttributes struct {
+	Cadence         *string `url:"cadence,omitempty" json:"cadence,omitempty"`
+	KeepN           *int    `url:"keep_n,omitempty" json:"keep_n,omitempty"`
+	OlderThan       *string `url:"older_than,omitempty" json:"older_than,omitempty"`
+	NameRegex       *string `url:"name_regex,omitempty" json:"name_regex,omitempty"` // NameRegex will be replaced by NameRegexDelete but now calls without setting this field are not accepted.
```

Comment:
What value needs to be set here? The same as used with `NameRegexDelete`? AS if that is the case it will be very unclear for users of this API to know that they have to specify the same value twice.

Maybe we should set the value ourselves to allow users to ignore this field. What do you think? Could be something like adding this to the function bodies:

```
if opt.ContainerExpirationPolicyOptions != nil {
  opt.ContainerExpirationPolicyOptions.NameRegex = opt.ContainerExpirationPolicyOptions.NameRegexDelete
}
```

Reply:
I would test if `opt.ContainerExpirationPolicyOptions.NameRegex == nil` before assigning `opt.ContainerExpirationPolicyOptions.NameRegexDelete` to it but it sounds like a good idea.

Reply:
I added a method to do this since it's used in three different places. Please let me know what you think of the solution.